src/Mauro/Greeter.ts

```diff
@@ -1,10 +1,15 @@
 export class Greeter {
-  constructor() {}
+  private _getCurrentTime: Date;
+  constructor(currentTime: Date) {
+    this._getCurrentTime = currentTime;
```

Comment:
Take care here.

`currentTime` seems to be a constant but `_getCurrentTime` seems to be a function (at least in the name).

Also, if `currentTime` is a constant, what happens if the greeter instance lives for a long time?